src/node/server/serverPluginModuleRewrite.ts

```diff
@@ -199,7 +202,7 @@ export function rewriteImports(
         }
       }
 
-      if (hasHMR) {
+      if (hasHMR && enableHMR) {
```

Comment:
Maybe you'd prefer the `enableHMR` check in the `hasHMR` declaration instead?

Reply:
Ah, this is what breaks `react-refresh`... Looks like I have more work ahead of me. 😬